src/renderers/dom/client/eventPlugins/SimpleEventPlugin.js

```diff
@@ -634,7 +656,7 @@ var SimpleEventPlugin = {
   },
 
   willDeleteListener: function(inst, registrationName) {
-    if (registrationName === 'onClick') {
+    if (registrationName === 'onClick' && isInteractive(inst._tag)) {
```

Comment:
Oof... Yes. Looks like there's also no test coverage here. I'll add that as well.